src/components/TokenEnterAmount.tsx

```diff
@@ -177,66 +179,94 @@ export function useEnterAmount(props: {
         : ''
 
     const parsedTokenAmount = parseInputAmount(convertedLocalToToken, decimalSeparator)
+    const balanceInLocal = convertTokenToLocalAmount({
+      tokenAmount: props.token.balance,
+      tokenInfo: props.token,
+      usdToLocalRate,
+    })?.decimalPlaces(2)
+
+    const isMaxLocalAmount =
+      parsedLocalAmount && balanceInLocal && balanceInLocal.eq(parsedLocalAmount)
+    const tokenAmount = isMaxLocalAmount ? props.token.balance : parsedTokenAmount
 
     return {
       token: {
-        amount: convertedLocalToToken,
-        bignum: parsedTokenAmount,
-        displayAmount: getDisplayTokenAmount(parsedTokenAmount, props.token),
+        bignum: tokenAmount,
+        displayAmount: getDisplayTokenAmount(tokenAmount, props.token),
       },
       local: {
-        amount: parsedLocalAmount?.toFixed(2) ?? '',
         bignum: parsedLocalAmount,
         displayAmount: getDisplayLocalAmount(parsedLocalAmount, localCurrencySymbol),
       },
     }
-  }, [amountRaw, amountType, localCurrencySymbol])
+  }, [amount, amountType, localCurrencySymbol])
 
   function handleToggleAmountType() {
-    setAmountType((prev) => (prev === 'local' ? 'token' : 'local'))
+    const newAmountType = amountType === 'local' ? 'token' : 'local'
```

Comment:
@kathaypacific considering that the state setting is indeed batched since React 18 and the whole scope of `handleToggleAmountType` will happen only once at the very end - it is exactly what I need. I need these two states to be updated at the same time (and only once) and batching ensures that.

And [according to React docs](https://react.dev/learn/queueing-a-series-of-state-updates):
_React does not batch across multiple intentional events like clicks—each click is handled separately. _

So even if user clicks multiple time - we will be sure that both states will always be in sync with one another as it will only execute everything in scope as a single update.

P.S. I do remember that they also whether had this stated at the react docs previously or it was just some Abramov's post but there was a statement that using state setter without callback is completely fine if the use-case is simple enough and it will benefit from batching (which I believe applies in this case).